app/scripts/metamask-controller.js

```diff
@@ -2173,22 +2175,28 @@ export default class MetamaskController extends EventEmitter {
         ),
       // msg signing
       ///: BEGIN:ONLY_INCLUDE_IF(build-main,build-beta,build-flask)
-      processTypedMessage:
-        this.signatureController.newUnsignedTypedMessage.bind(
-          this.signatureController,
-        ),
-      processTypedMessageV3:
-        this.signatureController.newUnsignedTypedMessage.bind(
-          this.signatureController,
-        ),
-      processTypedMessageV4:
-        this.signatureController.newUnsignedTypedMessage.bind(
-          this.signatureController,
-        ),
-      processPersonalMessage:
-        this.signatureController.newUnsignedPersonalMessage.bind(
-          this.signatureController,
-        ),
+
+      processTypedMessage: (...args) =>
+        addSignatureMessage({
+          SignatureController: this.signatureController,
+          signatureParams: args,
+          type: MESSAGE_TYPE.ETH_SIGN_TYPED_DATA,
+        }),
+      processTypedMessageV3: addSignatureMessage({
```

Comment:
Should these three be `() => ` as currently they are evaluating immediately and assigning the result of the function call?

Reply:
Yes, I forgot adding all leftovers 🤦